pkgs/by-name/al/alvr/package.nix

```diff
@@ -1,34 +1,154 @@
-{ lib,
-  appimageTools,
-  fetchurl,
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  substituteAll,
+  nix-update-script,
+  pkg-config,
+  autoAddDriverRunpath,
+  alsa-lib,
+  brotli,
+  bzip2,
+  celt,
+  ffmpeg,
+  jack2,
+  lame,
+  libX11,
+  libXi,
+  libXrandr,
+  libXcursor,
+  libdrm,
+  libglvnd,
+  libogg,
+  libpng,
+  libtheora,
+  libunwind,
+  libva,
+  libvdpau,
+  libxkbcommon,
+  openssl,
+  openvr,
+  pipewire,
+  rust-cbindgen,
+  soxr,
+  vulkan-headers,
+  vulkan-loader,
+  wayland,
+  x264,
+  xvidcore,
 }:
-let
+
+rustPlatform.buildRustPackage rec {
   pname = "alvr";
-  version = "20.6.1";
-  src = fetchurl {
-    url = "https://github.com/alvr-org/ALVR/releases/download/v${version}/ALVR-x86_64.AppImage";
-    hash = "sha256-IYw3D18xUGWiFu74c4d8d4tohZztAD6mmZCYsDNxR+A=";
+  version = "20.11.0";
+
+  src = fetchFromGitHub {
+    owner = "alvr-org";
+    repo = "ALVR";
+    rev = "refs/tags/v${version}";
+    fetchSubmodules = true; #TODO devendor openvr
+    hash = "sha256-zqeh9U0A/KHlRieq9Lf+7f04K3JG/vpE2gZ916ReXLc=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
```

Comment:
Is there any reason why we vendor `Cargo.lock`? I just tried building it with the following, and it worked just fine.

```nix
  cargoLock = {
    lockFile = "${src}/Cargo.lock";
    outputHashes = {
      "openxr-0.19.0" = "sha256-bnMSjJh+zjLw4Pdxr7LLm6qYAJOK7hz5xORKZ2pVcGw=";
      "settings-schema-0.2.0" = "sha256-luEdAKDTq76dMeo5kA+QDTHpRMFUg3n0qvyQ7DkId0k=";
    };
  };
```

Reply:
Well, that solves that! :)